server/src/main/java/org/elasticsearch/cluster/metadata/IndexMetadata.java

```diff
@@ -2141,6 +2141,11 @@ IndexMetadata build(boolean repair) {
                 // to be able to build a temporary instance
                 tierPreference = null;
             }
+            if (SearchableSnapshotsSettings.isPartialSearchableSnapshotIndex(settings)
+                && (tierPreference == null || tierPreference.size() != 1 || tierPreference.get(0).equals(DataTier.DATA_FROZEN) == false)) {
```

Comment:
If for some reason `tierPreference` is an empty list (hopefully never, but you never know), I think this would throw an `ArrayIndexOutOfBounds` exception. Can we add something to protect against that?

Reply:
The `tierPreference` will have one element by the time we evaluate the `get(0)` call because of the previous `||` condition is
```
tierPreference.size() != 1
```

Reply:
what about `tierPreference.size() > 1`?

Reply:
>what about tierPreference.size() > 1

I'd rather have the paranoid `!= 1` which gives us protection against an empty list (especially given where this code lives)

Reply:
Yep, I read it wrong, you're right!